### PR TITLE
Thread service grpc

### PR DESCRIPTION
**Pull request by sarunnut1236**
A lean version of thread service providing gRPC

### CODE REVIEW

File: thread-service/Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ RUN npm install
 COPY . .
 
 # Build the TypeScript code
-RUN npm run build
+RUN npx prisma generate & npm run build
```
Comment (ThanatWongsamut) [inline thread-service/Dockerfile]: There is no build command in `package.json`, did you tested the Dockerfile

File: thread-service/package.json [hunk 2]
```diff
@@ -2,50 +2,23 @@
...
+  "main": "index.js",
   "scripts": {
-    "dev": "tsx watch --clear-screen=false src/index.ts | pino-pretty",
-    "build": "tsup",
-    "start": "node dist/index.js",
-    "lint": "biome check src/",
-    "lint:fix": "biome check src/ --fix",
-    "format": "biome format src/ --fix"
+    "start": "npx prisma generate & npx ts-node src/server.ts",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (ThanatWongsamut) [inline thread-service/package.json:8]: Can remove this

File: thread-service/src/server.ts [hunk 3]
```diff
@@ -1,34 +1,208 @@
...
+    },
 
-export { app, logger };
+    createThread: async (
+        call: ServerUnaryCall<Thread, Thread>,
+        callback: sendUnaryData<Thread>
+    ) => {
+        try {
+            const thread = await prisma.thread.create({
+                data: call.request,
```
Comment (ThanatWongsamut) [inline thread-service/src/server.ts]: Do we need to validate the data first, or we can ignore?

File: thread-service/src/decorator.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+export const prepareUpdateThread = (
+    thread: Partial<Thread>
+): Partial<Thread> => {
+    const { id, ...updatedThread } = thread;
+    
+    updatedThread.updatedAt = new Date();
+    
+    if (updatedThread.createdAt) {
+        delete updatedThread.createdAt;
```
Comment (ThanatWongsamut) [inline thread-service/src/decorator.ts:11]: I think we can left createdAt (don't need to delete it)

**Review by ThanatWongsamut — COMMENTED** [review]
Left comment

File: thread-service/src/decorator.ts (hunk 4 above)
Comment (sarunnut1236) [reply thread-service/src/decorator.ts:11]: I think it's immutable once it's created

Comment (ThanatWongsamut) [reply thread-service/src/decorator.ts:11]: And why we need to delete it?

File: thread-service/prisma/schema.prisma [hunk 5]
```diff
@@ -1 +1,27 @@
...
+  previewFeatures = ["fullTextSearch"]
+}
+
+datasource db {
+  provider = "mongodb"
+  url      = env("DATABASE_URL")
+}
+
+model Thread {
+  id          String   @id @default(auto()) @map("_id") @db.ObjectId
```
Comment (ThanatWongsamut) [inline thread-service/prisma/schema.prisma]: Can you make it threadId instead, so we can differentiate from replyId or otherId

**Review by ThanatWongsamut — COMMENTED** [review]
More comment

File: thread-service/Dockerfile (hunk 1 above)
Comment (sarunnut1236) [reply thread-service/Dockerfile]: deleted haha

File: thread-service/src/server.ts (hunk 3 above)
Comment (sarunnut1236) [reply thread-service/src/server.ts]: data validated

File: thread-service/Dockerfile (hunk 1 above)
Comment (sarunnut1236) [reply thread-service/Dockerfile]: Edited

**Review by ThanatWongsamut — APPROVED** [review]

**Pull request merged**